hugegraph-test/src/main/java/com/baidu/hugegraph/core/IndexLabelCoreTest.java

```diff
@@ -322,6 +325,64 @@ public void testAddIndexLabelByNotBelongedProperty() {
         });
     }
 
+    @Test
+    public void testCollectionIndex() {
+        super.initPropertyKeys();
+        SchemaManager schema = graph().schema();
+        schema.propertyKey("tags").asText().valueSet().create();
+        schema.propertyKey("category").asText().valueSet().create();
+        schema.propertyKey("country").asText().create();
+
+        schema.vertexLabel("soft").properties("name", "tags", "country", "category")
+              .primaryKeys("name").create();
+
+        schema.indexLabel("softByTag").onV("soft").secondary()
+              .by("tags").create();
+
+        schema.indexLabel("softByCategory").onV("soft").search()
+              .by("category").create();
+
+        HugeGraph graph = graph();
+
+        graph.addVertex(T.label, "soft", "name", "hugegraph",
+                        "country", "china",
+                        "category", Arrays.asList("graphdb", "db"),
+                        "tags", Arrays.asList("graphdb", "gremlin"));
+
+        graph.addVertex(T.label, "soft", "name", "neo4j",
+                        "country", "usa",
+                        "category", Arrays.asList("graphdb", "db"),
+                        "tags", Arrays.asList("graphdb", "cypher"));
+
+        graph.addVertex(T.label, "soft", "name", "janusgraph",
+                        "country", "usa",
+                        "category", Arrays.asList("graphdb", "db"),
+                        "tags", Arrays.asList("graphdb", "gremlin"));
+
+        graph.tx().commit();
+
+        List<Vertex> vertices;
+        vertices = graph.traversal().V().has("soft", "category",
+                                             ConditionP.textContains("graphdb")).toList();
+        Assert.assertEquals(3, vertices.size());
+
+        // by single item
+        vertices = graph.traversal().V().has("soft", "tags", "gremlin").toList();
```

Comment:
the semantics of "has(key, value)" is "equals", because it is confusing,  prefer to don't allow this querying method.